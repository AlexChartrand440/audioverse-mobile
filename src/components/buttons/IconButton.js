import React from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/Feather'

const IconButton = ({ onPress, style, iconStyle, ...props }) => (
  <TouchableOpacity onPress={onPress} style={style}>
    <Icon {...props} style={iconStyle} />
  </TouchableOpacity>
)

IconButton.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.any,
  iconStyle: PropTypes.any
}

export default IconButton
