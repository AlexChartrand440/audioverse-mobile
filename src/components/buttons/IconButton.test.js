import React from 'react'
import IconButton from './IconButton'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(<IconButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
