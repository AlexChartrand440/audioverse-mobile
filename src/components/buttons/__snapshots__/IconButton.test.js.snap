// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        undefined,
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
